### PR TITLE
Make exception handling more robust for background updates and release v0.7.3

### PR DESCRIPTION
**Pull request by d00rman**
Bug: [T104581](https://phabricator.wikimedia.org/T104581)

### CODE REVIEW

File: lib/db.js [hunk 1]
```diff
@@ -588,7 +588,11 @@ DB.prototype._backgroundUpdates = function(req, limit, indexes) {
     // matching row, in descending order.
     return self._getRaw(newerRebuildRequest)
     .then(function(res) {    // Query for one record previous
-        return P.each(res.items.reverse(), handler.handleRow.bind(handler));
+        return P.try(function() {
+            return P.each(res.items.reverse(), handler.handleRow.bind(handler));
+        }).catch(function() {
+            // noop here for now
```
Comment (Pchelolo) [inline lib/db.js]: Could add logging here, at least we would know smth happened.

**Comment by Pchelolo** [comment]
LGTM, except one suggestion about logging.

File: lib/db.js (hunk 1 above)
Comment (d00rman) [inline lib/db.js]: Done

**Comment by d00rman** [comment]
I have released v0.7.2 in order to be able to re-deploy the last version of RESTBase with only the new version of the storage module.

**Comment by d00rman** [comment]
v0.7.3 has been released to npm, https://gerrit.wikimedia.org/r/223297 is the deployment candidate only bumping the version

File: lib/dbutils.js [hunk 2]
```diff
@@ -122,14 +122,16 @@ dbu.makeSchemaHash = function makeSchemaHash(schema) {
...
+        return client.execute_p(query, params, {
+            prepare: true,
+            fetchSize: options.fetchSize || 5,
+            pageState: pageState,
+        });
     })
     .catch(function(err) {
         if (!options.retries) {
-            throw err;
+            return P.reject(err);
```
Comment (gwicke) [inline lib/dbutils.js]: Afaik this is equivalent to throw, as it is inside a promise handler.

File: lib/dbutils.js [hunk 3]
```diff
@@ -150,19 +152,29 @@ function _nextPage(client, query, params, pageState, options) {
...
         return _nextPage(client, query, params, pageState, options)
         .then(function(res) {
-            return P.each(res.rows, handler)
-            .then(function() {
-                if (res.pageState === null) {
+            return P.try(function() {
+                return P.each(res.rows, handler);
+            }).catch(function(e) {
+                // something fishy's going on, abort abort abort!
+                return P.reject(e);
```
Comment (gwicke) [inline lib/dbutils.js]: Same here; you are basically re-raising the error.

**Comment by gwicke** [comment]
LGTM overall, just two small cases where I think a .catch is not needed / doesn't make a difference.

**Comment by d00rman** [comment]
@gwicke forgot to remove these useless (and desperate) changes. Removed them.

**Pull request merged**